cmd/build/v2/services.go

```diff
@@ -86,31 +87,28 @@ func (bc *OktetoBuilder) GetServicesToBuild(ctx context.Context, manifest *model
 	return svcsToBuildList, nil
 }
 
-// checkServiceToBuild looks for the service image reference at the registry and adds it to the buildCh if is not found
-func (bc *OktetoBuilder) checkServiceToBuild(service string, manifest *model.Manifest, buildCh chan string) error {
+// checkServiceToBuildDuringDeploy looks for the service image reference at the registry and adds it to the buildCh
+// if is not found. This function is called during deploy operations (up, deploy, destroy and compose) to check if
+// images have to be built or not. In that case, we only check the existence of "okteto" tag in the dev registry
+func (bc *OktetoBuilder) checkServiceToBuildDuringDeploy(service string, manifest *model.Manifest, buildCh chan string) error {
 	buildInfo := manifest.Build[service].Copy()
 	isStack := manifest.Type == model.StackType
 	if isStack && bc.oktetoContext.IsOkteto() && !bc.Registry.IsOktetoRegistry(buildInfo.Image) {
 		buildInfo.Image = ""
 	}
 
-	var err error
-	var buildHash string
-	if bc.smartBuildCtrl.IsEnabled() {
-		buildHash, err = bc.smartBuildCtrl.GetBuildHash(buildInfo)
-		if err != nil {
-			bc.ioCtrl.Logger().Infof("error getting build hash: %s", err)
-		}
-	}
-
-	imageChecker := getImageChecker(buildInfo, bc.Config, bc.Registry, bc.smartBuildCtrl, bc.ioCtrl.Logger())
-	imageWithDigest, err := imageChecker.getImageDigestReferenceForService(manifest.Name, service, buildInfo, buildHash)
+	imageChecker := getImageChecker(bc.Config, bc.Registry, bc.smartBuildCtrl, bc.ioCtrl.Logger())
+	imageWithDigest, err := imageChecker.getImageDigestReferenceForServiceDeploy(manifest.Name, service, buildInfo)
 	if oktetoErrors.IsNotFound(err) {
 		bc.ioCtrl.Logger().Debug("image not found, building image")
 		buildCh <- service
 		return nil
 	} else if err != nil {
-		return err
+		oktetoLog.Warning("could not verify if image for service %s is already in the registry. Building image...", service)
+		// If there is an error trying to get the image from the registry, we just rebuild that image
```

Comment:
We should use the new logger already injected in the oktetoBuilder
```suggestion
		bc.ioCtrl.Out().Warning("could not verify if image for service %s is already in the registry. Building image...", service)
```

Reply:
Changed! I had to add the `Warning` function to the IOController but it wasn't yet